ctdcal/odf_io.py

```diff
@@ -214,6 +217,95 @@ def create_multi_index(df,index=['SSSCC','GPSLAT','GPSLON','CTDPRS']):
     df = df.set_index(index,drop=True)
     return df
 
+
+def _salt_loader(ssscc, salt_dir):
+    """
+    Load raw salt file into DataFrame and calculate salinity.
+    """
+    saltpath = salt_dir + ssscc  # salt files have no extension
+    with open(saltpath, newline="") as f:
+        saltF = csv.reader(
+            f, delimiter=" ", quoting=csv.QUOTE_NONE, skipinitialspace="True"
+        )
+        saltArray = []
+        for row in saltF:
+            saltArray.append(row)
+        del saltArray[0]  # remove header
+
+    header = OrderedDict(  # having this as a dict streamlines next steps
+        [
+            ("STNNBR", int),
+            ("CASTNO", int),
+            ("SAMPNO", int),
+            ("BathTEMP", int),
+            ("CRavg", float),
+            ("autosalSAMPNO", int),
+            ("Unknown", int),
+            ("StartTime", object),
+            ("EndTime", object),
+            ("Attempts", int),
+        ]
+    )
+    saltDF = pd.DataFrame.from_records(saltArray)
+    # add as many "Reading#"s as needed
+    for ii in range(0, len(saltDF.columns) - len(header)):
+        header["Reading{}".format(ii + 1)] = float
+    saltDF.columns = list(header.keys())  # name columns
+    saltDF = saltDF[saltDF["autosalSAMPNO"] != "worm"]
+    saltDF = saltDF.astype(header)  # force dtypes
+    saltDF["SALNTY"] = gsw.SP_salinometer(
+        (saltDF["CRavg"] / 2.0), saltDF["BathTEMP"]
+    ).round(4)
+    return saltDF
+
+
+def _salt_exporter(saltDF, outdir="data/salt/", stn_col="STNNBR", cast_col="CASTNO"):
+    """
+    Export salt DataFrame to .csv file. Extra logic is included in the event that
+    multiple stations and/or casts are included in a single raw salt file.
+    """
+    stations = saltDF[stn_col].unique()
+    for station in stations:
+        stn_salts = saltDF[saltDF[stn_col] == station]
+        casts = stn_salts[cast_col].unique()
+        for cast in casts:
+            stn_cast_salts = stn_salts[stn_salts[cast_col] == cast]
+            outfile = (  # format to SSSCC_salts.csv
+                outdir + "{0:03}".format(station) + "{0:02}".format(cast) + "_salts.csv"
+            )
+            if not Path(outfile).exists():
+                stn_cast_salts.to_csv(outfile, index=False)
+            else:
+                print(outfile + " already exists...skipping")
```

Comment:
I'd probably reverse tests like this and short circuit the control flow, no idea if this is idiomatic python

```suggestion
            if Path(outfile).exists():
                print(outfile + " already exists...skipping")
                continue
            stn_cast_salts.to_csv(outfile, index=False)
```